Issue 963 endpoint for users group

garaimanoj opened the pull request:
![image](https://github.com/user-attachments/assets/0bb52692-00ba-492f-8e9b-09882437c3a8)
![image](https://github.com/user-attachments/assets/1f635e67-4c87-4b53-bc80-3766f632252a)

garaimanoj committed:
Add endpoints to retrieve users groups

garaimanoj committed:
Add license

garaimanoj committed:
Add tese cases

garaimanoj committed:
Test get user groups for subgroup

garaimanoj committed:
Replace the type specification with the diamond operator

garaimanoj committed:
Add joining date and scope policy description

garaimanoj committed:
Format date and label

enricovianello committed:
Bump to version 1.13.0

S203932 committed:
Include Subject DN field when searching for a user (#952)

S203932 committed:
Include Subject DN field when searching for a user (#952)

garaimanoj committed:
Store user affiliation into user info table (#781)

* Display affiliation in user profile

garaimanoj committed:
Store user affiliation into user info table (#781)

* Display affiliation in user profile

federicaagostini committed:
Return only FQANs with parent group equal to the VO name in the proxy (#993)

garaimanoj committed:
Add endpoint to get the list of clients owned by a user (#974)

mmutadich committed:
Option to limit custom scopes to admins only (#994)

mmutadich committed:
Option to limit custom scopes to admins only (#994)

rmiccoli committed:
Allow users to remove themselves from groups (#960)

rmiccoli committed:
Allow users to remove themselves from groups (#960)

mmutadich committed:
SCIM API bulk operations (#969)

mmutadich committed:
Configurable support button (#998)

mmutadich committed:
Configurable support button (#998)

federicaagostini committed:
Unlink certificate using both subject and issuer DNs (#1008)

Added issuer to the related AUDIT event
Delete duplicated and unused event class

federicaagostini committed:
Unlink certificate using both subject and issuer DNs (#1008)

Added issuer to the related AUDIT event
Delete duplicated and unused event class

S203932 committed:
Implement SCIM filtering (#946)

Supported operators: eq, co
Supported attributes: familyName, givenName, username, active, emails

S203932 committed:
Implement SCIM filtering (#946)

Supported operators: eq, co
Supported attributes: familyName, givenName, username, active, emails

enricovianello commented on iam-login-service/src/main/java/it/infn/mw/iam/api/account/group/AccountGroupController.java:
```diff
@@ -42,14 +51,43 @@
+  @GetMapping("/iam/account/me/groups")
+  public ListResponseDTO<RegisteredGroupDTO> getMyGroups(@Validated PaginatedRequestForm form,
+      final BindingResult validationResult) {
+    IamAccount account = accountUtils.getAuthenticatedUserAccount()
```
Probably we could directly call getUsersGroups(account.getId(), form, validationResult)

garaimanoj replied:
Done

enricovianello commented on iam-login-service/src/main/java/it/infn/mw/iam/api/common/RegisteredGroupDTO.java:
```diff
@@ -0,0 +1,177 @@
+import it.infn.mw.iam.persistence.model.IamScopePolicy;
+
+public class RegisteredGroupDTO implements Serializable {
+    private Long id;
```
strange indentation. I'd expect 2 spaces and not 4.

garaimanoj replied:
Done

enricovianello commented on iam-login-service/src/main/java/it/infn/mw/iam/api/common/RegisteredGroupDTO.java:
```diff
@@ -0,0 +1,177 @@
+                        .name(parentGroup.getName())
+                        .description(parentGroup.getDescription())
+                        .parentGroup(parentGroup.getParentGroup())
+                        // .childrenGroups(parentGroup.getChildrenGroups())
```
remove comment

garaimanoj replied:
Done

enricovianello commented on iam-login-service/src/main/java/it/infn/mw/iam/api/common/RegisteredGroupDTO.java:
```diff
@@ -0,0 +1,177 @@
+                            .uuid(gr.getUuid())
+                            .name(gr.getName())
+                            .description(gr.getDescription())
+                            //.parentGroup(gr.getParentGroup())
```
remove comment

garaimanoj replied:
Done

garaimanoj committed:
Fix indentation and remove comment

garaimanoj committed:
Remove duplicate code

garaimanoj committed:
Merge branch 'develop' into issue-963-endpoint-for-users-group

darcato committed:
Enable command line execution for development (#825)

* The CONTRIBUTING file has been updated properly
* Enabled also for test-client and VOMS-aa
---------
Co-authored-by: Federica Agostini <federica.agostini@cnaf.infn.it>

rmiccoli committed:
Test acr_values is added when mfa profile is enabled (#1022)

DonaldChung-HK committed:
Ensure access tokens are not issued before totp verification

This is to fix the security issue where user with OIDC and SAML login can have their token issued without typing in TOTP. This is because the principle for issuring token is the client and so are not restricted by the `PRE_AUTHENTICATED` role assigned to the user as it just check if the user is authenticated. The issue is that OIDC and SAML can't be given a `Authenticated = false` since it violates the contract with spring `ProviderManager`. This caused the mitreID to issue the token despite user not finishing authentication and a under only a partial authentication. Thus, this commit aims to address the above by:
- A check is added to the `createAccessToken` Method so that an exception is triggered when client asking for access token to be issued to a user that contains the role `PRE_AUTHENTICATED`.
- A test that confirms if the subject has the role  `PRE_AUTHENTICATED`  an access token will not be issued and an exception is raised.

enricovianello committed:
Update CHANGELOG.md

enricovianello committed:
Explicit declaration of Bouncy Castle dependency (#1023)

Explicit dependency declaration to avoid any conflicts with saml2-core and mitreId dependencies
Imported from INDIGO IAM v1.12.1

enricovianello committed:
Explicit declaration of Bouncy Castle dependency (#1023)

Explicit dependency declaration to avoid any conflicts with saml2-core and mitreId dependencies
Imported from INDIGO IAM v1.12.1

S203932 committed:
Configurable text in the registration button (#997)

S203932 committed:
Configurable text in the registration button (#997)

enricovianello committed:
Add validation when registration fields are read-only (#1024)

* Remove unused and useless password from registration DTO
---------
Co-authored-by: Federica Agostini <federica.agostini@cnaf.infn.it>

enricovianello committed:
Upgrades from release v1.12.2 (#1026)

* Move to voms-api-java to 3.3.6 which means canl 2.8.3 and bc 1.81
* Update CHANGELOG.md
---------
Co-authored-by: Federica Agostini <federica.agostini@cnaf.infn.it>

mmutadich committed:
Add backward offset for 'nbf' claim (#1017)

Added a configurable backward offset for 'nbf' claim in order to avoid time discrepancies.

rmiccoli committed:
Avoid Group Managers to access account's search API (#1018)

enricovianello committed:
Add RPM build workflow on Continuous Integration (#1027)

Moved rpm stuff of iam-login-service into a proper directory.
Build for el8 and el9.
Platform el10 needs maven compiler 3.11 or newer, expected after the migration to latest Spring Security libs.

enricovianello committed:
Add RPM build workflow on Continuous Integration (#1027)

Moved rpm stuff of iam-login-service into a proper directory.
Build for el8 and el9.
Platform el10 needs maven compiler 3.11 or newer, expected after the migration to latest Spring Security libs.

enricovianello committed:
Fix unresolved variable on GitHub workflow and typo

enricovianello committed:
Fix unresolved variable on GitHub workflow and typo

enricovianello committed:
Use almalinux instead of redhat as RPM target directory

enricovianello committed:
Use almalinux instead of redhat as RPM target directory

garaimanoj committed:
Merge branch 'develop' into issue-963-endpoint-for-users-group

garaimanoj committed:
Remove duplicate tag
